tests/yam/migration/setup_upgrade_env.pm

```diff
@@ -22,14 +22,23 @@ sub run {
             get_var('SCC_ADDONS')));
 
     # Save the original value of the variables in order to restore it later if needed
-    set_var('VERSION_ENV', get_var('VERSION'));
-    set_var('SCC_ADDONS_ENV', get_var('SCC_ADDONS'));
+    foreach my $var (qw(AGAMA SCC_ADDONS VERSION)) {
+        set_var($var . "_ENV", get_var($var)) if (get_var($var));
+    }
 
     # Change variables to the other version that we want to migrate from/to
-    set_var('VERSION', $version);
-    record_info('VERSION', 'VERSION=' . get_var('VERSION'));
-    set_var('SCC_ADDONS', $scc_addons);
-    record_info('SCC_ADDONS', 'SCC_ADDONS=' . get_var('SCC_ADDONS'));
+    my $agama = '0';
```

Comment:
we should read the value instead of setting it, I guess better would be to have:
`AGAMA => get_var('AGAMA')` wdyt?